coding-201-advanced-parsers/metaparser.py

```diff
@@ -0,0 +1,197 @@
+                if 'neighbor' not in PYTHON_DICT['vrf'][vrf_name]:
+                    PYTHON_DICT['vrf'][vrf_name]['neighbor'] = {}
+                if neighbor_name not in PYTHON_DICT['vrf'][vrf_name]\
+                    ['neighbor']:
+                    PYTHON_DICT['vrf'][vrf_name]['neighbor'][neighbor_name] = {}
+                if 'address_family' not in PYTHON_DICT['vrf'][vrf_name]\
+                    ['neighbor'][neighbor_name]:
+                    PYTHON_DICT['vrf'][vrf_name]['neighbor'][neighbor_name]\
+                        ['address_family'] = {}
+                if address_family_name not in PYTHON_DICT['vrf'][vrf_name]\
+                    ['neighbor'][neighbor_name]['address_family']:
+                    PYTHON_DICT['vrf'][vrf_name]['neighbor'][neighbor_name]\
+                        ['address_family'][address_family_name] = {}
+
+                if direction == 'in':
+                    PYTHON_DICT['vrf'][vrf_name]['neighbor'][neighbor_name]\
+                        ['address_family'][address_family_name]\
+                            ['route_map_in'] = route_map_name
+                else:
+                    PYTHON_DICT['vrf'][vrf_name]['neighbor'][neighbor_name]\
+                        ['address_family'][address_family_name]\
+                            ['route_map_out'] = route_map_name
+
+                continue
+
+        # print the parsed built structure
+        pprint.pprint(PYTHON_DICT)
+
+# Expected output
+# ===============
+
+# PYTHON_DICT = {
+#     'vrf':
+#         {'VRF1':
+#             {'neighbor':
+#                 {'21.0.0.1':
+#                     {'address_family':
+#                         {'VPNv4 Unicast':
+#                             {'route_map_in': 'RMAP',
+#                              'route_map_out': 'RMAP2'}
+#                         }
+#                     }
+#                 }
+#             },
+#         'VRF2':
+#             {'neighbor':
+#                 {'21.0.0.2':
+#                     {'address_family':
+#                         {'VPNv6 Unicast':
+#                             {'route_map_in': 'RMAP3',
+#                              'route_map_out': 'RMAP4'}
+#                         }
+#                     },
+#                 '21.0.0.4':
+#                     {'address_family':
+#                         {'VPNv6 Unicast':
+#                             {'route_map_in': 'RMAP7',
+#                              'route_map_out': 'RMAP8'}
+#                         }
+#                     }
+#                 }
+#             },
+#         'default':
+#             {'neighbor':
+#                 {'21.0.0.3':
+#                     {'address_family':
+#                         {'VPNv4 Unicast':
+#                             {'route_map_in': 'RMAP5',
+#                              'route_map_out': 'RMAP6'}
+#                         }
+#                     }
+#                 }
+#             }
+#         }
+#     }
```

Comment:
if __name__ = '__main__':
    # give an example of running it

Reply:
Done.